File: plugins/devkit/devkit-java-tests/testData/inspections/cancellationCheckInLoops/RunConditionOnSuperMethod.java
package inspections.cancellationCheckInLoops;

import com.intellij.util.concurrency.annotations.RequiresReadLock;

import static inspections.cancellationCheckInLoops.Foo.doSomething;

abstract class Bar {

  @RequiresReadLock
  public abstract void bar();

}


class BarImpl extends Bar {

  @Override
  public void bar() {
    String[] items = {""};
    <warning descr="Cancellation check 'com.intellij.openapi.progress.ProgressManager.checkCanceled' should be the first statement in a loop body">for</warning> (String item : items) {
      doSomething();
    }
  }

}

File: plugins/devkit/devkit-kotlin-tests/testData/inspections/cancellationCheckInLoops/RunConditionOnSuperMethod.kt
package inspections.cancellationCheckInLoops

import com.intellij.util.concurrency.annotations.RequiresReadLock

import inspections.cancellationCheckInLoops.Foo.doSomething


abstract class Bar {

  @RequiresReadLock
  abstract fun bar()

}

class BarImpl : Bar() {

  override fun bar() {
    <warning descr="Cancellation check 'com.intellij.openapi.progress.ProgressManager.checkCanceled' should be the first statement in a loop body">for</warning> (i in 1..10) {
      doSomething()
    }
  }

}